src/main/java/org/apache/sling/scripting/core/impl/jsr223/SlingScriptEngineManager.java

```diff
@@ -72,8 +76,21 @@
         policy = ReferencePolicy.DYNAMIC
     )
 )
+@Designate(ocd=SlingScriptEngineManager.Config.class)
 public class SlingScriptEngineManager extends ScriptEngineManager implements BundleListener {
 
+    @ObjectClassDefinition(name ="Apache Sling Script Engine Manager",
+            description="Configures options for the Script Engine Manager")
+    public @interface Config {
+
+        @AttributeDefinition(name = "Includes", description = "Regular expressions to match script engines to include (matched against short names).")
+        String[] includes() default {".*"};
+
+        @AttributeDefinition(name = "Excludes", description = "Optional regular expressions to match script engines to exclude (matched against short names). Allows to exclude specific engines from selected set as produced by 'Includes'.\"")
```

Comment:
I'd say "A script engine with a short name that matches any of these (optional) expressions is NOT activated, even if it was accepted by the 'Includes' configuration" - I think it's easier to understand.